semgrep-core/Matching/Generic_vs_generic.ml

```diff
@@ -598,11 +599,11 @@ and m_expr a b =
    * const a = "foo"; ... a == "foo" would be catched by $X == $X.
   *)
   | A.L(a1), b1 ->
-      (match Normalize_generic.constant_propagation_and_evaluate_literal b1 with
-       | Some b1 ->
-           m_literal_constness a1 b1
-       | None -> fail ()
-      )
+      if_config (fun x -> x.Config.constant_propagation)
```

Comment:
@IagoAbal we can now enable/disable if we want.

Reply:
Remains to hook this up in the rule format to let user enable/disable equivalences/features.

Reply:
Great!